### PR TITLE
Add the AWS managed LOG4JRCE mitigation

### PR DESCRIPTION
**Pull request by johnjcsmith**
Add the Log4jRCE mitigation rule from `AWSManagedRulesKnownBadInputsRuleSet`.

### CODE REVIEW

File: lib/waf.ts [hunk 1]
```diff
@@ -25,6 +25,30 @@ export class WebApplicationFirewall extends Construct {
...
+      priority: 0,
+      overrideAction: { none: {} },
+      statement: {
+          managedRuleGroupStatement: {
+              name: 'AWSManagedRulesKnownBadInputsRuleSet',
+              vendorName: 'AWS',
+              excludedRules: [
+                   {name: 'Host_localhost_HEADER'},
+                   {name: 'PROPFIND_METHOD'},
+                   {name: 'ExploitablePaths_URIPATH'}
```
Comment (hubertott) [inline lib/waf.ts:40]: Observation: You updated Line 139 to be upper-case as per the docs, but these are not?

Comment (johnjcsmith) [reply lib/waf.ts:40]: I copied these directly from the console list of rules in this rule set :shrug:

Comment (johnjcsmith) [reply lib/waf.ts:40]: Do the docs recommend capitalising?

Comment (hubertott) [reply lib/waf.ts:40]: I was referring to `name` vs `Name` but there seems to be a bit of inconsistency anway.

**Review by hubertott — APPROVED** [review]

**Review by crispy101 — APPROVED** [review]
LGTM, as tested in preprod.

**Pull request merged**